### PR TITLE
FB login

### LINKED ISSUE #255: Add GraphQL API to support sign in with Facebook account
Currently, users can only sign in through with their Github account. Some of the users may not use Github. Adding Facebook sign in option makes Short more accessible.

### PR DESCRIPTION
**Pull request by duderman**
Adds login with FB

New config options are expected:
* `FACEBOOK_CLIENT_ID` - contains `App id` for current app created at FB developers portal
* `FACEBOOK_CLIENT_SECRET` - contains `App Secret` value for the current FB App
* `FACEBOOK_REDIRECT_URI` - `redirect_uri` param for all the facebook calls containing a callback endpoint (_couldn't use web frontend URL as it needs to be redirected directly to the API endpoitn_)

Following endpoints were created:
* `GET /oauth/facebook/sign-in` for initiating auth process
* `GET /oauth/facebook/sign-in/callback` for FB oauth callback

FB login flow docs: https://developers.facebook.com/docs/facebook-login/manually-build-a-login-flow

Closes #255

### CODE REVIEW

**Comment by byliuyang** [comment]
Hi @duderman , thanks for the amazing work you did! Please address comment from code climate and add unit tests! Looking forwards to merge this PR!

File: README.md [hunk 1]
```diff
@@ -129,12 +142,16 @@ Visit [http://localhost:3000](http://localhost:3000)
    RECAPTCHA_SECRET=your_recaptcha_secret
    GITHUB_CLIENT_ID=your_Github_client_id
    GITHUB_CLIENT_SECRET=your_Github_client_secret
+   FACEBOOK_CLIENT_ID=your_facebook_app_id
```
Comment (byliuyang) [inline README.md]:
```suggestion
   FACEBOOK_CLIENT_ID=your_Facebook_app_id
```

File: README.md [hunk 2]
```diff
@@ -129,12 +142,16 @@ Visit [http://localhost:3000](http://localhost:3000)
...
+   FACEBOOK_CLIENT_SECRET=your_facebook_app_secret
```
Comment (byliuyang) [inline README.md]:
```suggestion
   FACEBOOK_CLIENT_SECRET=your_Facebook_app_secret
```

File: backend/app/adapter/facebook/api.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package facebook
+
+import (
+	"short/app/entity"
+
+	fb "github.com/huandu/facebook"
+)
+
+// API access user's profile information through FB Graph API.
```
Comment (byliuyang) [inline backend/app/adapter/facebook/api.go]:
```suggestion
// API accesses user's profile information through FB Graph API.
```

Comment (byliuyang) [reply backend/app/adapter/facebook/api.go]: I wonder is this Graph or GraphQL?

File: backend/dep/provider/oauth.go [hunk 4]
```diff
@@ -20,3 +20,22 @@ func NewGithubOAuth(
...
 	return oauth.NewGithub(req, string(clientID), string(clientSecret))
 }
+
+// FacebookClientID represents client ID used for Facebook OAuth.
+type FacebookClientID string
+
+// FacebookClientSecret represents client secret used for Facebook OAuth.
+type FacebookClientSecret string
+
+// FacebookRedirectURI represents redirect_uri for facebook
```
[the remark this answers is not in the document]
Comment (duderman) [reply backend/dep/provider/oauth.go]: they call it Facebook Login. I should probably phrase it the same way?

File: backend/app/adapter/facebook/api.go (hunk 3 above)
Comment (duderman) [reply backend/app/adapter/facebook/api.go]: I know it's super confusing but facebook calls their public API Graph API and it has nothing to do with GraphQL 😕

**Comment by duderman** [comment]
@byliuyang Addressed some of the comments above

PR for enabling the button is here: https://github.com/byliuyang/short/pull/294

File: README.md [hunk 5]
```diff
@@ -129,12 +143,16 @@ Visit [http://localhost:3000](http://localhost:3000)
...
+   FACEBOOK_CLIENT_SECRET=your_Facebook_app_secret
+   FACEBOOK_REDIRECT_URI=http://localhost/oauth/facebook/sign-in/callback
    JWT_SECRET=your_JWT_secret
    WEB_FRONTEND_URL=http://localhost:3000
    ```
-   
+
 1. Update `DB_HOST`, `DB_PORT`, `DB_USER`, `DB_PASSWORD`, `DB_NAME`,
-   `RECAPTCHA_SECRET`, `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `JWT_SECRET`
+   `RECAPTCHA_SECRET`, `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `JWT_SECRET`,\
```
Comment (byliuyang) [inline README.md]: I wonder whether `\` is necessary here.

**Pull request merged**